ngraph/core/include/ngraph/ngraph_visibility.hpp

```diff
@@ -25,11 +25,15 @@
 #pragma warning(disable : 4275)
 #endif
 
+#ifdef NGRAPH_STATIC_LIBRARY // defined if we are building or calling NGRAPH as a static library
+#define NGRAPH_API
+#else
 #ifdef ngraph_EXPORTS // defined if we are building the NGRAPH DLL (instead of using it)
```

Comment:
I think we need to rely on `ngraph_EXPORTS` - it's added by cmake automatically in case if target type is `SHARED` library.

Reply:
ngraph_EXPORTS is defined only when ngraph is being compiled in shared mode, so it should not be used in that case ?

Reply:
Will it work?
```cmake
#ifdef ngraph_EXPORTS // defined if we are building the NGRAPH DLL (instead of using it)
#define NGRAPH_API NGRAPH_HELPER_DLL_EXPORT
#else
#define NGRAPH_API
#endif // ngraph_EXPORTS
```

Reply:
In the case we are the caller of the dll, changing dllimport with nothing:
- might have an impact if some variable and not just functions are masked with NGRAPH_API
- will probably break the ABI

It's usually how i do in my own projects (using dllexport but not dllimport) but I don't know what side effect we might have here.

Reply:
@alalek I see that OpenCV does not have dllimport. Can absense of dllimport brake something in our case?

Reply:
As i expected because it failed to link all the global variables exported with dllexport (something that you might not have with OpenCV)

Such as in ngraph\core\include\ngraph\type\element_type.hpp
        extern NGRAPH_API const Type undefined;
        extern NGRAPH_API const Type dynamic;
        extern NGRAPH_API const Type boolean;
        extern NGRAPH_API const Type bf16;
        extern NGRAPH_API const Type f16;
        extern NGRAPH_API const Type f32;
        extern NGRAPH_API const Type f64;
        extern NGRAPH_API const Type i8;
        extern NGRAPH_API const Type i16;
        extern NGRAPH_API const Type i32;
        extern NGRAPH_API const Type i64;
        extern NGRAPH_API const Type u1;
        extern NGRAPH_API const Type u8;
        extern NGRAPH_API const Type u16;
        extern NGRAPH_API const Type u32;
        extern NGRAPH_API const Type u64;